swift/apple/FirezoneNetworkExtension/SessionManager.swift

```diff
@@ -0,0 +1,250 @@
+import FirezoneKit
+import Foundation
+import NetworkExtension
+
+/// SessionManager owns the Session lifecycle and manages command/event multiplexing.
+///
+/// This actor follows Android's pattern where the event loop owns the Session instance.
+/// When the event loop exits, the Session is automatically cleaned up, ensuring Drop
+/// is called on the Rust side to prevent memory leaks.
+///
+/// Commands are sent via AsyncStream (similar to Kotlin's Channel).
+/// Events are polled from Rust's nextEvent() and forwarded to the event handler.
+actor SessionManager {
+  // Session ownership
+  private var session: Session?
+  private var eventLoopTask: Task<Void, Never>?
+
+  // Command channel (Swift equivalent of Kotlin's Channel)
+  private let commandContinuation: AsyncStream<SessionCommand>.Continuation
+  private let commandStream: AsyncStream<SessionCommand>
+
+  // Event handler callback
+  private let eventHandler: (Event) async -> Void
+
+  // Logging interval for event loop health monitoring
+  private static let eventPollingLogInterval = 100
+
+  init(eventHandler: @escaping (Event) async -> Void) {
+    self.eventHandler = eventHandler
+    (commandStream, commandContinuation) = AsyncStream.makeStream()
+  }
```

Comment:
Instead of a callback, I think it would be nicer to also have an `AsyncStream` here.